build.go

```diff
@@ -46,6 +54,12 @@ func Build(entryResolver EntryResolver,
 	return func(context packit.BuildContext) (packit.BuildResult, error) {
 		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
 
+		var buildEnv BuildEnvironment
+		_, err := env.UnmarshalFromEnviron(&buildEnv)
+		if err != nil {
+			return packit.BuildResult{}, fmt.Errorf("failed to parse build configuration: %w", err)
+		}
+
```

Comment:
What if we moved this stuff out into `run/main.go`? You'd then be able to pass the `BuildEnvironment` struct into the `Build` function more easily for tests without needing an interface.

Reply:
Ah yes. That would be much nicer!

Reply:
@ryanmoran In the latest commit, I took up your suggestion. All in all, I feel pretty good about this approach. My one qualm is that the go-env library doesn't have the clearest error logging (e.g. if some env var can't be parsed, it will log the _value_ that couldn't be parsed, but not which env var was set to that value).

Reply:
@ForestEckhardt is going to experiment with using a different environment unmarshalling library and we'll compare notes.

Reply:
I've decided to use https://github.com/caarlos0/env.